test/components/interpolatedColorLegendTests.ts

```diff
@@ -3,113 +3,114 @@
+        legend.renderTo(svg);
+
+        assertBasicRendering(legend);
+
+        const legendElement: d3.Selection<void> = (<any> legend)._element;
+        const labels = legendElement.selectAll("text");
+        const swatchContainer = legendElement.select(".swatch-container");
+        const swatchContainerBCR = (<Element> swatchContainer.node()).getBoundingClientRect();
+
+        const lowerLabelBCR = (<Element> labels[0][0]).getBoundingClientRect();
+        const upperLabelBCR = (<Element> labels[0][1]).getBoundingClientRect();
+        assert.closeTo(lowerLabelBCR.left, swatchContainerBCR.left, window.Pixel_CloseTo_Requirement, "left align first label ");
+        assert.closeTo(upperLabelBCR.right, swatchContainerBCR.right, window.Pixel_CloseTo_Requirement, "right align second label");
+        if (orientation === "top") {
+          assert.operator(lowerLabelBCR.top, "<=", swatchContainerBCR.top, `first label is to the top of swatches`);
+          assert.operator(upperLabelBCR.top, "<=", swatchContainerBCR.top, `second label is to the top of swatches`);
+        } else {
+          assert.operator(lowerLabelBCR.bottom, ">=", swatchContainerBCR.bottom, `first label is to the bottom of swatches`);
+          assert.operator(upperLabelBCR.bottom, ">=", swatchContainerBCR.bottom, `second label is to the bottom of swatches`);
+        }
+        assert.operator(upperLabelBCR.left, ">=", lowerLabelBCR.right, "lower label is drawn to the left of upper label");
+        svg.remove();
+      });
+    });
 
-    svg.remove();
+    VERITCAL_ORIENTATIONS.forEach((orientation: string) => {
+      it(`renders swatches and labels in correct order when oriented ${orientation}`, () => {
+        const legend = new Plottable.Components.InterpolatedColorLegend(colorScale);
+        legend.orientation(orientation);
+        legend.renderTo(svg);
+
+        assertBasicRendering(legend);
+
+        const legendElement: d3.Selection<void> = (<any> legend)._element;
+        const labels = legendElement.selectAll("text");
+        const swatchContainer = legendElement.select(".swatch-container");
+        const swatchContainerBCR = (<Element> swatchContainer.node()).getBoundingClientRect();
+
+        const lowerLabelBCR = (<Element> labels[0][0]).getBoundingClientRect();
+        const upperLabelBCR = (<Element> labels[0][1]).getBoundingClientRect();
+        assert.closeTo(lowerLabelBCR.bottom, swatchContainerBCR.bottom, window.Pixel_CloseTo_Requirement, "bottom align first label");
+        assert.closeTo(upperLabelBCR.top, swatchContainerBCR.top, window.Pixel_CloseTo_Requirement, "top align second label");
+        if (orientation === "left") {
+          assert.operator(lowerLabelBCR.left, "<=", swatchContainerBCR.left, `first label is to the left of swatches`);
+          assert.operator(upperLabelBCR.left, "<=", swatchContainerBCR.left, `second label is to the left of swatches`);
```

Comment:
Shouldn't we check that the `right` of the labels is `<=` the `left` of the swatch container?